std/math.d

```diff
@@ -8659,6 +8659,10 @@ private real polyImpl(real x, in real[] A) @trusted pure nothrow @nogc
    Computes whether a values is approximately equal to a reference value,
    admitting a maximum relative difference, and a maximum absolute difference.
 
+   Warning:
+        This template is considered out-dated. It will be removed from
```

Comment:
Do we really want to remove `approxEqual`?
It seems like a commonly used and low-maintenance function, I think we can keep it around undocumented for backwards compatibility.

Reply:
It's the last step of a long road. It started with [issue15881](https://issues.dlang.org/show_bug.cgi?id=15881), where someone complained that `approxEqual` was not doing, what he expected to do. I had the same experience some months before I read that issue and did some research on this issue and came up with one of my first PRs. 

The reason for replacing `approxEqual` is mainly, that it's like a security belt that doesn't hold, when it is needed. Think of this: You wrote a function, added a unittest, and because with `approxEqual` and felt safe. But later, someone else changed something and your function didn't work anymore as expected. Still, the unittest didn't notice. And why? Because the test with `approxEqual` was too weak.

The main idea of this function is to overcome the problems which arise, when comparing floatingpoint numbers with ==. This doesn't work out well due to rounding errors. So `approxEqual` should accept two numbers as equal, when they only differ by some rounding errors. But acctually `approxEqual` accepts also numbers as equal, that are far off compared to rounding errors.

Having said all this: IMHO it's time to remove `approxEqual` completely, because else people feel safe, where they aren't, probably without knowing.

Reply:
> Think of this: You wrote a function, added a unittest, and because with approxEqual and felt safe.

That's a fair point, but I'm also thinking of this:
You're learning D, you find a code snippet using approxEqual, (e.g. [this](http://rosettacode.org/wiki/Temperature_conversion#D)), it doesn't compile because of `approxEqual`, and you find no trace of what happened to it in the documentation, and you get confused.

Maybe we can replace the function body of `approxEqual` with a `static assert(0, "use isClose instead")` once the time comes to remove it.

Reply:
Ideally, until the deprecation period we will update any (external) docs that use `approxEqual`.